Rumba pin error?

The pin assignment for the Rumba board don't corespond to neither of the schematics I have found for the BIQU Tango v1.0 and Rumba board which have identical pin outs. So I can't assign the correct pins for my TMC2130 drivers.

I have a tango board. FAN_PIN is pin 7 and FAN1_PIN is pin 8.  The printouts on the board do label these as Fan0 and Fan1.

This is different than the pins_rumba.h where Fan_PIN is pin 8 and Fan1_PIN 7

In the past I just swapped out the two fans at the board, but leaving this issue unreported can lead to overheating or thermal runaway for newbies.

So I recently I reverted the wires back to see if I can code my way out of this.

Attempt 1:
Configuration_adv.h
#define E0_AUTO_FAN_PIN 7
error: "You cannot set E0_AUTO_FAN_PIN equal to FAN_PIN.

Attempt 2:
Configuration_adv.h
#define FAN_PIN 8 
#define E0_AUTO_FAN_PIN 7
error: 
sketch\src\gcode\control\M42.cpp: In static member function 'static void GcodeSuite::M42()':
sketch\src\gcode\control\M42.cpp:62:9: error: duplicate case value

         case FAN1_PIN: thermalManager.fan_speed[1] = pin_status; break;

Attempt 3: 
pins_Rumba.h
#ifndef FAN_PIN
  #define FAN_PIN           8
#endif
#ifndef FAN1_PIN //  <-------------------------- I added this to stop the pins_Rumba.h from overwriting my value
  #define FAN1_PIN            7
#endif

Attempt 4:
Configuration_adv.h
#define FAN_PIN             8
#define FAN1_PIN            -1   // <---- This not possible with current version
#define E0_AUTO_FAN_PIN 7

pins_Rumba.h
#ifndef FAN_PIN
  #define FAN_PIN           7
#endif
#ifndef FAN1_PIN //  <-------------------------- I added this to stop the pins_Rumba.h from overwriting my value
  #define FAN1_PIN            8
#endif

As it stands I can't build Marlin with as is.  I need to add #ifndef FAN_1PIN clause so I can overwrite Configuration_adv.h with :
#define FAN_PIN             8
#define FAN1_PIN            -1   // <---- This not possible with current version
#define E0_AUTO_FAN_PIN 7

Is there any way we can have a pins_Tango.h board for a few of these changes?  If not, can we have the above change to add an ifndef for FAN1_PIN?

I have made sandbox changes but next time I overwrite, these changes will be lost.  Any thoughts?

--- a/Marlin/src/core/boards.h
+++ b/Marlin/src/core/boards.h
@@ -99,6 +99,7 @@
 #define BOARD_OVERLORD                1143  // Overlord/Overlord Pro
 #define BOARD_HJC2560C_REV1           1144  // ADIMLab Gantry v1
 #define BOARD_HJC2560C_REV2           1145  // ADIMLab Gantry v2
+#define BOARD_TANGO                   1146  // BIQU Tango V1
 
 //
 // RAMBo and derivatives

--- a/Marlin/src/pins/pins.h
+++ b/Marlin/src/pins/pins.h
@@ -180,6 +180,8 @@
   #include "ramps/pins_Z_BOLT_X_SERIES.h"       // ATmega2560                             env:megaatmega2560
 #elif MB(TT_OSCAR)
   #include "ramps/pins_TT_OSCAR.h"              // ATmega2560                             env:megaatmega2560
+#elif MB(TANGO)
+  #include "ramps/pins_TANGO.h"                 // ATmega2560                             env:megaatmega2560
 
 //
 // RAMBo and derivatives

--- a/Marlin/src/pins/ramps/pins_RUMBA.h
+++ b/Marlin/src/pins/ramps/pins_RUMBA.h
@@ -162,7 +162,7 @@
   #define SPINDLE_LASER_PWM_PIN 4   // Hardware PWM. Pin 4 interrupts OC0* and OC1* always in use?
 #endif
 #ifndef SPINDLE_LASER_ENA_PIN
-  #define SPINDLE_LASER_ENA_PIN    14   // Pullup!
+  #define SPINDLE_LASER_ENA_PIN 14   // Pullup!
 #endif
 #ifndef SPINDLE_DIR_PIN
   #define SPINDLE_DIR_PIN  15

--- a/Marlin/src/pins/ramps/pins_RUMBA.h
+++ b/Marlin/src/pins/ramps/pins_RUMBA.h
@@ -143,8 +143,9 @@
 #ifndef FAN_PIN
   #define FAN_PIN           7
 #endif
-#define FAN1_PIN            8
-
+#ifndef FAN1_PIN
+  #define FAN1_PIN            8
+#endif
 //
 // Misc. Functions
 //

--- a/Marlin/src/pins/ramps/pins_RUMBA.h
+++ b/Marlin/src/pins/ramps/pins_RUMBA.h
@@ -143,7 +143,7 @@
 #ifndef FAN_PIN
   #define FAN_PIN           7
 #endif
-#ifndef FAN1_PIN
+#if !defined(FAN1_PIN)
   #define FAN1_PIN            8
 #endif
 //

--- a/Marlin/src/pins/ramps/pins_RUMBA.h
+++ b/Marlin/src/pins/ramps/pins_RUMBA.h
@@ -143,9 +143,10 @@
 #ifndef FAN_PIN
   #define FAN_PIN           7
 #endif
-#if !defined(FAN1_PIN)
-  #define FAN1_PIN            8
+#ifndef FAN1_PIN
+  #define FAN1_PIN          8
 #endif
+
 //
 // Misc. Functions
 //

--- a/Marlin/src/pins/ramps/pins_TANGO.h
+++ b/Marlin/src/pins/ramps/pins_TANGO.h
@@ -27,15 +27,12 @@
 
 #define BOARD_INFO_NAME "Tango"
 
- #define FAN_PIN           8
- #define FAN1_PIN          -1
- 
- #if  !defined(E0_AUTO_FAN_PIN) || (E0_AUTO_FAN_PIN==-1)
-   #undef E0_AUTO_FAN_PIN
-   #define E0_AUTO_FAN_PIN 7
- #endif
+#define FAN_PIN             8
+#define FAN1_PIN           -1
 
- #ifndef TEMP_0_PIN
+#define ORIG_E0_AUTO_FAN_PIN 7
+
+#ifndef TEMP_0_PIN
   #if TEMP_SENSOR_0 == -1
     #define TEMP_0_PIN     10   // Analog Input (connector *K1* on Tango thermocouple ADD ON is used)
   #else
@@ -51,4 +48,4 @@
   #endif
 #endif
 
- #include "pins_RUMBA.h"
+#include "pins_RUMBA.h"

--- a/Marlin/src/pins/ramps/pins_TANGO.h
+++ b/Marlin/src/pins/ramps/pins_TANGO.h
@@ -1,0 +1,54 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (c) 2019 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (c) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#pragma once
+
+/**
+ * BIQU Tango pin assignments
+ */
+
+#define BOARD_INFO_NAME "Tango"
+
+ #define FAN_PIN           8
+ #define FAN1_PIN          -1
+ 
+ #if  !defined(E0_AUTO_FAN_PIN) || (E0_AUTO_FAN_PIN==-1)
+   #undef E0_AUTO_FAN_PIN
+   #define E0_AUTO_FAN_PIN 7
+ #endif
+
+ #ifndef TEMP_0_PIN
+  #if TEMP_SENSOR_0 == -1
+    #define TEMP_0_PIN     10   // Analog Input (connector *K1* on Tango thermocouple ADD ON is used)
+  #else
+    #define TEMP_0_PIN     15   // Analog Input (default connector for thermistor *T0* on rumba board is used)
+  #endif
+#endif
+
+#ifndef TEMP_1_PIN
+  #if TEMP_SENSOR_1 == -1
+    #define TEMP_1_PIN      9   // Analog Input (connector *K2* on Tango thermocouple ADD ON is used)
+  #else
+    #define TEMP_1_PIN     14   // Analog Input (default connector for thermistor *T1* on rumba board is used)
+  #endif
+#endif
+
+ #include "pins_RUMBA.h"

